### ISSUE TITLE
ranger fails to build if pytest and flake8 are not installed

### ISSUE THREAD

**Issue by eliasdaler** [ranger/ranger#1213]
Hello.

Today I've tried to build ranger on PC where basically very few Python modules were installed. But first, the build failed because pytest wasn't installed and then it failed because of missing flake8.

Can you please add these dependencies in the building script somehow to install these modules automatically if they're missing?

**Comment by Vifon** [comment]
Very valid points though I'd rather create a proper `requirements.txt` file with all the dependencies and change the default make target.

Guys, why is `test` called in the default target? It's confusing. It's true `make install` should be enough for most people but let's keep the make targets consistent with the rest of the world. I think it should run only `compile`.

**Comment by toonn** [comment]
I disagree. First of all the instructions are in the readme and they don't say to blindly do `make && sudo make install`. The default target is for developer convenience imo, it's still a makefile not an "installer."

That being said I invoke specific test targets anyway so I don't care much either way but it's not as if just `compile` is all that useful a default target either.

**Comment by eliasdaler** [comment]
Oh, I see. If I just do "sudo make install", it installs just fine without pytest and flake8 installed. Maybe it's just a habit of doing "make && make install" when building things manually. :)

Still, maybe it will be useful for developers to say in README that pytest and flake8 are needed if they want to run tests.

**Comment by toonn** [comment]
Habits are hard to break. The info on the dependencies for testing is in `HACKING.md` : )
If we do anything about this I'd make the default target a no-op so `make && sudo make install` works just as expected. Not convinced it's necessary though, I expect the number of users manually installing ranger with the makefile to be tiny.

**Comment by Vifon** [comment]
@toonn, what do you think about making `help` the default target?

**Comment by toonn** [comment]
As long as it's a no-op in spirit, I'm fine with it.
